app/addons/components/react-components.react.jsx

```diff
@@ -1133,13 +1141,20 @@ define([
         return null;
       }
       return (
-        <i className="icon fonticon-ok-circled" />
+        <i className={"icon " + this.props.customIcon} />
       );
     },
 
     render: function () {
       return (
-        <button onClick={this.props.onClick} type="submit" className="btn btn-success save" id={this.props.id}>
+        <button
+          onClick={this.props.onClick}
+          type="submit"
+          data-id={this.props['data-id']}
+          className={'btn save ' + this.props.buttonType}
+          id={this.props.id}
+          style={this.props.style}
+        >
```

Comment:
Couple of minor things:
- ID isn't a required prop, so it may end up blank. Better to only add the prop if defined.
- data-id doesn't have default prop value.